### PR TITLE
sql: Add `SHOW`

### LINKED ISSUE #1955: sql: implement SHOW var_name

### PR DESCRIPTION
**Pull request by tamird**
Closes #1955.

### CODE REVIEW

File: sql/show.go [hunk 1]
```diff
@@ -20,11 +20,36 @@ package sql
...
+		v.rows = append(v.rows, []parser.Datum{parser.DString(p.session.Database)})
+	case `SYNTAX`:
+		switch parser.Syntax(p.session.Syntax) {
+		case parser.Modern:
+			v.rows = append(v.rows, []parser.Datum{parser.DString("modern")})
+		case parser.Traditional:
+			v.rows = append(v.rows, []parser.Datum{parser.DString("traditional")})
+		}
+	default:
+		return nil, util.Errorf("unknown variable: %s", name)
```
Comment (petermattis) [inline sql/show.go]: I think this error needs a test. For errors returned to the user, we usually use `fmt.Errorf` as the line number isn't terribly useful to the end user.

**Comment by petermattis** [comment]
LGTM

File: sql/show.go (hunk 1 above)
Comment (tamird) [inline sql/show.go]: Ah, you're right. This was pasta from set.go.

**Comment by MBerhault** [comment]
LGTM

File: sql/show.go (hunk 1 above)
Comment (petermattis) [inline sql/show.go]: That shouldn't use `util.Errorf` either. Mind fixing it as well?

Comment (tamird) [inline sql/show.go]: fixed

**Comment by tamird** [comment]
added stringer for parser.Syntax to reduce boilerplate. might want to have another look

File: sql/show.go (hunk 1 above)
Comment (tamird) [inline sql/show.go]: yep fixed and added a test

File: sql/parser/parse.go [hunk 2]
```diff
@@ -28,6 +28,7 @@ import (
 	"errors"
 )
 
+//go:generate stringer -type=Syntax
```
Comment (petermattis) [inline sql/parser/parse.go]: Put this right next to the `Stringer` type declaration.

File: sql/set.go [hunk 3]
```diff
@@ -51,17 +50,17 @@ func (p *planner) Set(n *parser.Set) (planNode, error) {
...
-		case "modern":
+		switch normalizeName(string(s)) {
+		case normalizeName(parser.Modern.String()):
 			p.session.Syntax = int32(parser.Modern)
-		case "traditional":
+		case normalizeName(parser.Traditional.String()):
 			p.session.Syntax = int32(parser.Traditional)
 		default:
-			return nil, fmt.Errorf("%s: \"%s\" is not in (\"modern\", \"traditional\")", name, s)
+			return nil, fmt.Errorf("%s: \"%s\" is not in (%q, %q)", name, s, parser.Modern.String(), parser.Traditional.String())
```
Comment (petermattis) [inline sql/set.go]: No need for the explicit `.String()` calls.

Comment (petermattis) [inline sql/set.go]: And any reason for using `%q` here? `%s` seems more appropriate.

Comment (tamird) [inline sql/set.go]: thanks, copy pasta

**Comment by petermattis** [comment]
LGTM

File: sql/set.go (hunk 3 above)
Comment (tamird) [inline sql/set.go]: re `%q`: i thought we'd want the quotes

File: sql/parser/parse.go (hunk 2 above)
Comment (tamird) [inline sql/parser/parse.go]: done

File: sql/set.go (hunk 3 above)
Comment (petermattis) [inline sql/set.go]: Ok.

**Pull request merged**